From f7a86775d6040b02c97f44df9f8f3cb3ddb03ddc Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Fri, 20 Apr 2018 09:07:20 +0100
Subject: [PATCH 2/4] qemu: Turn shared directory setup into one long command

If we run this as four separate commands, then we'll get four shell
prompts, but because we only wait for the first, we'll get out of sync.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 virt/autopkgtest-virt-qemu | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/virt/autopkgtest-virt-qemu b/virt/autopkgtest-virt-qemu
index 57a7478..6f3fef2 100755
--- a/virt/autopkgtest-virt-qemu
+++ b/virt/autopkgtest-virt-qemu
@@ -220,9 +220,9 @@ def setup_shared(shared_dir):
 
     term = VirtSubproc.get_unix_socket(os.path.join(workdir, 'ttyS1'))
 
-    term.send(b'''mkdir -p -m 1777 /run/autopkgtest/shared
-mount -t 9p -o trans=virtio,access=any autopkgtest /run/autopkgtest/shared
-chmod 1777 /run/autopkgtest/shared
+    term.send(b'''mkdir -p -m 1777 /run/autopkgtest/shared &&
+mount -t 9p -o trans=virtio,access=any autopkgtest /run/autopkgtest/shared &&
+chmod 1777 /run/autopkgtest/shared &&
 touch /run/autopkgtest/shared/done_shared
 ''')
 
-- 
2.17.0

